File: sdks/guides/ruby/integrations/embedding_openai.rb
# frozen_string_literal: true

# Third-party integration - requires: bundle add openai
# See: /guides/rag-and-indexing

require "openai"

OpenAI::Client.new

# > OpenAI embedding usage
def embed(text)
  response = client.embeddings(parameters: { model: "text-embedding-3-small", input: text })
  response.dig("data", 0, "embedding") || []
end
# !!

File: sdks/guides/typescript/integrations/embedding-openai.ts
// Third-party integration - requires: pnpm add openai
// See: /guides/rag-and-indexing

import OpenAI from 'openai';

const client = new OpenAI();

// > OpenAI embedding usage
export async function embed(text: string): Promise<number[]> {
  const r = await client.embeddings.create({
    model: 'text-embedding-3-small',
    input: text,
  });
  return r.data[0]?.embedding ?? [];
}
// !!
